app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers;
+use App\Http\Requests\StoreCategoryRequest;
+use App\Http\Requests\UpdateCategoryRequest;
+use App\Services\CategoryService;
+use Illuminate\Http\Request;
+
+class CategoryController extends Controller
+{
+    protected $categoryService;
+
+    // Constructor to inject CategoryService
+    public function __construct(CategoryService $categoryService)
+    {
+        $this->categoryService = $categoryService;
+    }
+
+    // Get all categories
+    public function index()
+    {
+        $categories = $this->categoryService->getAllCategories();
+        return response()->json($categories);
+    }
+
+    // Get a category by ID
+    public function show($id)
```

Comment:
Please use model route binding. Model route binding allows Laravel to automatically resolve the model instance from the route parameter, making the code cleaner and more readable. It also handles 404 errors automatically if the model is not found, improving the application's reliability and user experience.